test/Dialect/FIRRTL/errors.mlir

```diff
@@ -903,3 +903,127 @@ firrtl.circuit "DupSymField" {
     %w3 = firrtl.wire sym [<@x,1,public>] : !firrtl.vector<uint<1>,1>
   }
 }
+
+// -----
+// Check upward reference XMRs
+
+firrtl.circuit "func2" {
+  firrtl.module @func2(in %ref_in1: !firrtl.ref<uint<1>>, out %ref_out: !firrtl.ref<uint<1>>) {
+    // expected-error @+1 {{connect has invalid flow for ref type ports: the source expression "ref_in1" and destination expression "ref_out" both have same port kind, expected Module port to Instance connections only}}
```

Comment:
Nit: this is a "kind" error and not a flow error. Also, the terminology of source is wrong. Source is a flow which is a function of the type, and other stuff, while operations, specifically connects, have "operands" or LHS/RHS.